### ISSUE TITLE
The detection accuracy of the R-50-FPN Mask R-CNN is lower than reference 

### ISSUE THREAD

**Issue by qwerty-dotcom** [pytorch/vision#1391]
Hi @fmassa , thanks for all the super helpful code.
I am trying to use the detection script in: https://github.com/pytorch/vision/tree/master/references/detection to train R-50-FPN Mask R-CNN.
When I re-train the detection AP is only 35.5, which should be 36.8 according to the reference model zoo.

I am using:
python 3.6.8
torch version 1.2.0
torchvision version 0.4.0
CUDA version 10.0.130

I have only 6 GPU and I am allocating one image per GPU. I have changed the learning rate and warm up iteration as suggested in Detectron: lr = 0.0075 and warmup iterations = 1333. 
I am starting from Imagenet pretrained (pretrain_backbone=True)

I am using the following :
!CUDA_VISIBLE_DEVICES=0,1,2,3,4,5 python3 -m torch.distributed.launch --nproc_per_node=6 --use_env train.py --lr=0.0075 --batch-size=6 --epochs 26 --lr-steps 16 22 --aspect-ratio-group-factor 3

Namespace(aspect_ratio_group_factor=3, 
	  batch_size=6, 
	  dataset='coco', 
          device='cuda', 
	  dist_backend='nccl', 
	  distributed=True, 
	  epochs=26, 
	  gpu=0, 
	  lr=0.0075, 
	  lr_gamma=0.1, 
	  lr_step_size=8, 
	  lr_steps=[16, 22], 
	  model='maskrcnn_resnet50_fpn', 
	  momentum=0.9)
I am a bit confused on what should I do to replicate the numbers.
Can you help me?
Thanks ! ^^

**Comment by fmassa** [comment]
This is weird.

I've reproduced the results using the commands from https://github.com/pytorch/vision/issues/1140#issuecomment-513386987

```python
python -m torch.distributed.launch --nproc_per_node=8 --use_env train.py --dataset coco --model maskrcnn_resnet50_fpn --epochs 26 --lr-steps 16 22 --aspect-ratio-group-factor 3
```
and took the best model (which was at epoch 22)

I also noted that you changed the global batch size, as you are using 6 imgs per GPU for 6 GPUs (total batch size of 36), while we trained our models on 2 imgs per GPU on 8 GPUs, which is a batch size of 16, so it looks like the lr scaling rules that you applied are not exactly the same.

Can you maybe try using 4 imgs / gpu and 4 GPUs, with the other parameters kept the same?

**Comment by qwerty-dotcom** [comment]
Thx for the help! It was indeed a problem regarding the number of GPU and the batch size ^^'.  Training with 4 GPU and 4 images per GPU worked.

**Comment by fmassa** [comment]
Cool, glad it worked
